doc/api/cli.md

```diff
@@ -156,6 +156,40 @@ Currently the support for run-time snapshot is experimental in that:
    a report in the [Node.js issue tracker][] and link to it in the
    [tracking issue for user-land snapshots][].
 
+### `--compile=path`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+> This feature is currently being designed and will still change.
+
+Compiles a JavaScript file into a single executable application.
+
+```console
+$ cat index.js
+console.log('Hello, world!');
+$ node --compile=cli index.js
+$ ./cli
+Hello, world!
+```
+
+This currently only supports compiling a CommonJS file into a Node.js binary
+that has any of the following executable formats:
+
+* Mach-O
+* PE
+* ELF
+
+Support for the following features are yet to come:
+
+* Compiling an ESM file
+* Compiling multiple files
+* Baking command line arguments
+* Compiling into XCOFF executable format
```

Comment:
```suggestion
* Compiling an ESM file.
* Compiling multiple files.
* Baking command line arguments.
* Compiling into XCOFF executable format.
```

Reply:
Thanks, added `.`s to the new list in the newly added docs.